util/buildRelease/gen_release

```diff
@@ -1,158 +1,82 @@
-        $commit = $ENV{'CHPL_GEN_RELEASE_COMMIT'};
-        print "[gen_release] Checking out revision $commit...\n";
-        SystemOrDie("cd $rootdir && git reset --hard $commit");
-    }
-
-    print "[gen_release] Confirm final Git source version used.\n";
-    SystemOrDie("cd $rootdir && git --no-pager status && git --no-pager log -1");
-
-
-
-    # write the git-sha while we have the .git directory
-    SystemOrDie("cd $rootdir && util/config/write-git-sha frontend/lib/util --build-version --chpl-home=$rootdir");
-    # use git archive to copy all the files under source control into the archive directory.
-    print "[gen_release] Creating build workspace with git archive...\n";
-    SystemOrDie("cd $rootdir && git archive --format=tar HEAD | (cd $archive_dir && tar -xf -)");
-    # explicitly copy the BUILD_VERSION file and git-version.cpp files
-    print "[gen_release] Copying BUILD_VERSION file.\n";
-    SystemOrDie("cp $rootdir/compiler/main/BUILD_VERSION $archive_dir/compiler/main/BUILD_VERSION");
-    print "[gen_release] Copying git-version.cpp\n";
-    SystemOrDie("cp $rootdir/frontend/lib/util/git-version.cpp $archive_dir/frontend/lib/util/git-version.cpp");
-
-    if (defined($ENV{"CHPL_HOME"})) {
-        $resultdir = $ENV{"CHPL_HOME"};
-        $resultdir = "$resultdir/tar";
-    } else {
-        $resultdir = $basetmpdir;
-    }
-}
+#!/usr/bin/env python3
+import os
+import sys
+import shutil
+import subprocess
+import tempfile
+import glob
+import itertools
+from contextlib import contextmanager
 
-# explicit files to include
-@files = (
-       "ACKNOWLEDGEMENTS.md",
-       "CHANGES.md",
-       "CONTRIBUTORS.md",
-       "COPYRIGHT",
-       "LICENSE",
-       "LICENSE.chapel",
-       "Makefile",
-       "CMakeLists.txt",
-       "Dockerfile",
-       "README.rst",
-       "README.files",
-       "compiler/codegen/reservedSymbolNames",
-       "configure",
-       "highlight/README.md",
-       "util/README",
-       "util/buildRelease/chpl-make-cpu_count",
-       "util/buildRelease/install.sh",
-       "util/chpl-completion.bash",
-       "util/printchplenv",
-       "util/setchplenv.bash",
-       "util/setchplenv.csh",
-       "util/setchplenv.fish",
-       "util/setchplenv.sh",
-       "util/start_test",
-       "util/chpltags",
-       "frontend/include/chpl/config/config.h.cmake"
-);
 
+@contextmanager
+def cd(path):
+    cwd = os.getcwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(cwd)
 
-# C/C++ sources
-@code_dirs = (
-    "compiler",
-    "frontend"
-);
 
+def run(*args, **kwargs):
+    print(f"+ {' '.join(args)}")
+    result = subprocess.run(args, **kwargs)
+    if result.returncode != 0:
+        raise RuntimeError(
+            f"[gen_release] Command failed with error code: {result.returncode}"
+        )
+    return result
+
+
+def make(*args, **kwargs):
+    return run("make", *args, **kwargs)
+
+
+def pjoin(*args):
+    return os.path.join(*args)
```

Comment:
why not `pjoin = os.path.join`?

or better yet, above, `from os.path import join`? I don't think you're using plain `join` anywhere.